### ISSUE TITLE
Large templates slows down shop template window

### ISSUE THREAD

**Issue by Illviljan** [Novynn/acquisitionplus#14]
My shop template has grown rather large, about 10k lines. It is mainly due to having a lot of these spoilers:
[spoiler="~b/o (X) (Currency)"]
{Stash:(X)(Currency_shorthand)}
[/spoiler]

where 
X = 1:0.5:100
Currency = ['chaos', 'regal', 'alteration', ..., 'mirror'] % Poe.trade currency name, all currencies that poe.trade supports. chaos, regal etc.
Currency_shorthand = ['c', 'regal', 'alt', ..., 'mirror'] % My preferred shorthand name of the currency, chaos = c, alteration = alt, etc.

This has worked really well, I've never needed to open acquisition plus and set any prices during the whole 1 month, I only changed the tab name in game.
However this has slowed down the template windows considerably, I suspect this can be done better with a new menu, defining currency, currency_shorthand and X in a nice and clear way.

**Comment by Novynn** [comment]
It seems like you're using the templates in quite an interesting way! It's intended that you set a buyout on the tab once then let the template system do its thing and sort it into spoilers for you (like [this](https://www.pathofexile.com/forum/view-thread/1440653)).

**Comment by Illviljan** [comment]
I can't access that link, I guess it's a beta only link. 
Anyways, here is my shop template:
http://pastebin.com/vdgF4wZ3

And here is my Matlab script to generate spoiler buyouts if someone wants to change shorthand currency names:
http://pastebin.com/HfvFNC3X

**Comment by Novynn** [comment]
Do you really have that many stash tabs? That's insane!

I can definitely see why it's being slow, it has a lot of template to parse. I can probably move it into a thread so it doesn't block the GUI but it would still take a while to complete.

**Comment by Illviljan** [comment]
I've got like 40 stash tabs. But I do change my buyouts on the fly depending on what I need at the moment, which is why I like the shop template so much. I did it this thoroughly to never have to redo the template again.

My idea was: IF "Alteration orb" has the shorthand name "alt" defined, THEN search for "alt" in the premium tabs and take the prefixed numbers before "alt" and use that as the buyout value. I'm no programmer so I'm not familiar how difficult this is to implement, but {stash:} almost works like that.
